api/server/httputils/form_test.go

```diff
@@ -111,22 +109,74 @@ func TestInt64ValueOrDefaultWithError(t *testing.T) {
 	}
 }
 
-func TestParsePlatformInvalid(t *testing.T) {
-	for _, tc := range []ocispec.Platform{
+func TestDecodePlatform(t *testing.T) {
+	tests := []struct {
+		doc          string
+		platformJSON string
+		expected     *ocispec.Platform
+		expectedErr  string
+	}{
+		{
+			doc:         "empty platform",
+			expectedErr: `failed to parse platform: unexpected end of JSON input`,
+		},
+		{
+			doc:          "not JSON",
+			platformJSON: `linux/ams64`,
+			expectedErr:  `failed to parse platform: invalid character 'l' looking for beginning of value`,
```

Comment:
FWIW; planning to make some improvements here after this;

- Friendlier messages for invalid / non-JSON
- Considering `os` to be required, but _possibly_ (TBD) `architecture` to be optional,
- :point_up: but: if a `variant` is specified, then architecture is required
- Considering `os.features` to be rejected; we currently don't use it for platform-matching, so it may be better to reject it until we do


Basically make the (from a CLI perspective) accepted format to be;

```console
<os>[(os.version)][/architecture|/architecture/variant]
```